DXMainClient/DXGUI/Multiplayer/CnCNet/ChoiceNotificationBox.cs

```diff
@@ -0,0 +1,189 @@
+
+namespace DTAClient.DXGUI.Multiplayer.CnCNet
+{
+    /// <summary>
+    /// A box that allows users to make a choice,
+    /// top-left of the game window.
+    /// </summary>
+    public class ChoiceNotificationBox : XNAPanel
+    {
+        const double DOWN_TIME_WAIT_SECONDS = 4.0;
+        const double DOWN_MOVEMENT_RATE = 2.0;
+        const double UP_MOVEMENT_RATE = 2.0;
+
+        public ChoiceNotificationBox(WindowManager windowManager) : base(windowManager)
+        {
+            downTimeWaitTime = TimeSpan.FromSeconds(DOWN_TIME_WAIT_SECONDS);
+        }
+
+        public Action<ChoiceNotificationBox> AffirmativeClickedAction { get; set; }
+        public Action<ChoiceNotificationBox> NegativeClickedAction { get; set; }
+
+        XNALabel lblHeader;
+        XNAPanel gameIconPanel;
+        XNALabel lblSender;
+        XNALabel lblChoiceText;
+        XNAButton affirmativeButton;
+        XNAButton negativeButton;
+
+        TimeSpan downTime = TimeSpan.Zero;
+
+        TimeSpan downTimeWaitTime;
+
+        bool isDown = false;
+
+        const int boxHeight = 101;
+
+        double locationY = -boxHeight;
+
+        public override void Initialize()
+        {
+            Name = "ChoiceNotificationBox";
+            BackgroundTexture = AssetLoader.CreateTexture(new Color(0, 0, 0, 196), 1, 1);
+            ClientRectangle = new Rectangle(0, -boxHeight, 300, boxHeight);
+            PanelBackgroundDrawMode = PanelBackgroundImageDrawMode.STRETCHED;
+
+            lblHeader = new XNALabel(WindowManager);
+            lblHeader.Name = "lblHeader";
+            lblHeader.FontIndex = 1;
+            lblHeader.AnchorPoint = new Vector2(ClientRectangle.Width / 2, 12);
+            lblHeader.TextAnchor = LabelTextAnchorInfo.CENTER;
+            lblHeader.Text = "MAKE A CHOICE";
+            AddChild(lblHeader);
+
+            gameIconPanel = new XNAPanel(WindowManager);
+            gameIconPanel.Name = "gameIconPanel";
+            gameIconPanel.ClientRectangle = new Rectangle(12, lblHeader.Bottom + 6, 16, 16);
+            gameIconPanel.DrawBorders = false;
+            gameIconPanel.BackgroundTexture = AssetLoader.TextureFromImage(ClientCore.Properties.Resources.dtaicon);
+            AddChild(gameIconPanel);
+
+            lblSender = new XNALabel(WindowManager);
+            lblSender.Name = "lblSender";
+            lblSender.FontIndex = 1;
+            lblSender.ClientRectangle = new Rectangle(gameIconPanel.Right + 3, lblHeader.Bottom + 6, 0, 0);
+            lblSender.Text = "fonger";
+            AddChild(lblSender);
+
+            lblChoiceText = new XNALabel(WindowManager);
+            lblChoiceText.Name = "lblChoiceText";
+            lblChoiceText.FontIndex = 1;
+            lblChoiceText.ClientRectangle = new Rectangle(12, lblSender.Bottom + 6, 0, 0);
+            lblChoiceText.Text = "What do you want to do?";
+            AddChild(lblChoiceText);
+
+            affirmativeButton = new XNAButton(WindowManager);
```

Comment:
Use `XNAClientButton` instead, this will save you from manually loading the texture.